R/createYaml.R

```diff
@@ -0,0 +1,161 @@
+      if (!is.null(v)) {
+        param$param_EV_PrecursorTolPPM = yc$setYAML("File$Evidence$MQpar_firstSearchTol_num", as.numeric(v))
+      }
+    }
+    
+    param$param_EV_PrecursorOutOfCalSD = yc$getYAML("File$Evidence$firstSearch_outOfCalWarnSD_num", param$param_EV_PrecursorOutOfCalSD)
+    
+    ## we do not dare to have a default, since it ranges from 6 - 4.5 ppm across MQ versions
+    param$param_EV_PrecursorTolPPMmainSearch = yc$getYAML("File$Evidence$MQpar_mainSearchTol_num", param$param_EV_PrecursorTolPPMmainSearch)
+    if (param$param_useMQPAR & !MZTAB_MODE && is.null(param)) {
+      v = getMQPARValue(txt_files$mqpar, "mainSearchTol") ## will also warn() if file is missing
+      if (!is.null(v)) {
+        param$param_EV_PrecursorTolPPMmainSearch = yc$setYAML("File$Evidence$MQpar_mainSearchTol_num", as.numeric(v))
+      }
+    }
+    if (is.na(param$param_EV_PrecursorTolPPMmainSearch))
+    {
+      warning("PTXQC: Cannot draw borders for calibrated mass error, since neither 'File$Evidence$MQpar_mainSearchTol_num' is set nor a mqpar.xml file is present!", immediate. = TRUE)
+    }
+    
+    param$param_MSMSScans_ionInjThresh = yc$getYAML("File$MsMsScans$IonInjectionThresh_num", param$param_MSMSScans_ionInjThresh, 0, 200)
+    
+    param$param_OutputFormats = yc$getYAML("PTXQC$OutputFormats", param$param_OutputFormats)
+    
+    param$param_PageNumbers = yc$getYAML("PTXQC$PlainPDF$AddPageNumbers", param$param_PageNumbers)
+    
+    
+    ####
+    ####  prepare the metrics
+    ####
+    lst_qcMetrics = PTXQC:::getMetricsObjects(DEBUG_PTXQC)
+    df.meta = PTXQC:::getMetaData(lst_qcMetrics = lst_qcMetrics)
+    df.meta
+    ## reorder metrics (required for indexing below!)
+    lst_qcMetrics_ord = lst_qcMetrics[df.meta$.id]
+    
+    ## write/update order from YAML
+    for (i in 1:nrow(df.meta))
+    {
+      #cat(paste("meta id: ", df.meta$.id[i], "\n"))
+      pname = paste0("order$", df.meta$.id[i])
+      pval = df.meta$order[i]
+      
+      ##check for shiny metrics input
+      if(!is.null(metrics)){
+        if(df.meta$.id[i] %in% metrics) yc$setYAML(pname, pval)
+        else yc$setYAML(pname, (-1))
+      }
+      
+      param_v = yc$getYAML(pname, pval)
+      ## update
+      if (is.numeric(param_v)) {
+        lst_qcMetrics_ord[[i]]$orderNr = param_v  # for some reason, lst_qcMetrics[[df.meta$.id]] does not work
+      } else {
+        stop("YAML param '", pname_v, "' is not numeric (", param_v, "). Please fix the YAML configuration!")
+      }
+    }
+    ## re-read meta (new ordering?)
+    df.meta = PTXQC:::getMetaData(lst_qcMetrics = lst_qcMetrics)
+    ## reorder metrics (again; after param update)
+    lst_qcMetrics_ord = lst_qcMetrics[df.meta$.id]
+    
+    ## write out the final YAML file (so users can disable metrics, if they fail)
+    yc$writeYAML(path)
+    
+    if(output) return(list(param, lst_qcMetrics_ord))

```

Comment:
always return a value, but make it named (see above)
`return list("param" = param ,. ...)`